### PR TITLE
CZGenEpi_Prep_PHB: implementing user-requested changes

### LINKED ISSUE #284: Make workflow to prepare metadata and files for CZGenEpi
:cool:

### :pushpin: Explain the Request

Make a workflow to prep metadata so it can be dropped directly into CZ GenEpi without any additional modifications.

### PR DESCRIPTION
**Pull request by sage-wright**
Closes #284 

## :hammer_and_wrench: Changes Being Made

As requested by users who demoed this workflow, the following changes have been implemented:
- an organism specification, since mpox and SARS-CoV-2 require different metadata parameters
- change the String input `sample_is_private_column_name` to be a Boolean `is_private`
- change the GISAID ID to be the GISAID Virus Name and generated from the `private_id` column automatically

### Impacted Workflows/Tasks

Workflows: `workflows/utilities/wf_czgenepi_prep.wdl`
Tasks: `tasks/utilities/task_czgenepi_wrangling.wdl`

## :brain: Context and Rationale

Automatically generating the GISAID virus name may lead to increased errors, but this trade-off may be worth it for labs with consistent and accurate formatting. 

## :clipboard: Workflow/Task Steps

Instead of extracting the `sample_is_private_column_name` and the `gisaid_id_column_name` columns, these columns are generated by the program using already-provided inputs and by the new `is_private` Boolean variable which is used to set the value for all samples in the set.

### Inputs

New inputs:
- `organism`: options are "sars-cov-2" (default) and "mpox"
- `is_private`: Boolean that if true (default) sets all samples to have "Yes" in the corresponding column (meaning private). Set to false for "No" (not private)

Removed inputs:
- `gisaid_column_name`
- `sample_is_private_column_name`

### Outputs

No new outputs have been created. 

### Impacted Outputs

- the field "GISAID ID (Public ID) - Optional" will now reflect the GISAID syntax for Virus Name. This will need to be checked for accuracy.

### Locally

The task works locally. Workflow cannot be tested locally due to Terra permissions.

### Terra

See test here: https://app.terra.bio/#workspaces/cdc-terra-resources/Theiagen_Wright_SC2_Sandbox/job_history/da52bac2-7931-42ed-aa70-25adbdbf1ac5

### Scenarios for Reviewer to Test

- cases with unusual sample names (many underscores, all numbers, etc.)

## :microscope: Quality checks

Pull Request (PR) checklist:
- [x] Include a description of what is in this pull request in this message.
- [x] The workflow/task has been tested locally and on Terra
- [x] The CI/CD has been adjusted and tests are passing
- [x] Everything follows the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-bb456f34322d4f4db699d4029050481c)

### CODE REVIEW

File: tasks/utilities/task_czgenepi_wrangling.wdl [hunk 1]
```diff
@@ -18,26 +18,57 @@ task czgenepi_wrangling {
...
+      r = re.compile('^\d{4}-\d{2}-\d{2}')      
+      if pd.isna(date):
+        print("Incorrect collection date format; collection date must be in YYYY-MM-DD format. Invalid date was: NaN")
+      elif r.match(date) is None:
+        print("Incorrect collection date format; collection date must be in YYYY-MM-DD format. Invalid date was: " + str(date))
+        return np.nan
+      else:
+        return date.split("-")[0]
+
+    if os.environ["is_private"] == "true":
```
Comment (kevinlibuit) [inline tasks/utilities/task_czgenepi_wrangling.wdl:52]: Will python error out if `is_private` isn't set? May be worth setting it default to false

File: tasks/utilities/task_czgenepi_wrangling.wdl [hunk 2]
```diff
@@ -18,26 +18,57 @@ task czgenepi_wrangling {
...
     # parse terra table for data
     python3 <<CODE
     import pandas as pd
     import numpy as np
     import re
     import os
 
+    # set a function to grab only the year from the date
+    def year_getter(date):
+      r = re.compile('^\d{4}-\d{2}-\d{2}')
```
Comment (kevinlibuit) [inline tasks/utilities/task_czgenepi_wrangling.wdl:43]: Not sure how much date filtering we care to include, but this seems to allow **invalid dates** that are in the _valid date format_, e.g. 2023-13-52

**Review by kevinlibuit — APPROVED** [review]
Code changes look solid; I'd wait on merge pending @emily-smith1 functional review

**Review by emily-smith1 — APPROVED** [review]
Tested successfully on 10 Mpox samples. The workflow ran as expected and the metadata and concatenated fasta files were successfully uploaded to CZGenEpi.

**Pull request merged**